### PR TITLE
Add flexibility in resizing method and algorithm used

### PR DESCRIPTION
**Pull request by SoapyDev**
I know this is a large pull request, but was not sure if this was still maintained.

Added -> ResizeType 
- Fit, work as usual
- Thumbnail for when you know the resize is gonna be smaller than the orginal. This is using a more efficient algorithm.
- Cover act as Object-fit: cover; in css, but at the resizing level. It will try to keep the format, but will remove the part of the image that do not fit.

Added ->  Filter
- CatmullRom, work as usual
- Nearest
- Laczos3 -> usualy a bit bigger than catmull, but they are also often the clearer
- Triangle
- Gaussian

The idea is to leave the user define what it needs the most, and what technique gives the best result.
Also, the Dynamic image is mutated in place, so we avoid creating a new one. 
This should reduce a bit the memory usage.

Added width and height to the <Image/> to reduce CLS.

### CODE REVIEW

File: src/image.rs [hunk 1]
```diff
@@ -9,17 +9,23 @@ use leptos_meta::Link;
...
     width: u32,
     /// Image quality. 0-100.
     #[prop(default = 75_u8)]
     quality: u8,
+    /// Filter type for the conversion : Nearest, Triangle, CatmullRom, Gaussian, Lanczos3
+    #[prop(default = "catmullrom")]
+    filter: &'a str,
+    /// Resize type for the conversion : Fit, Cover, Thumbnail
+    #[prop(default = "fit")]
+    resize_type: &'a str,
```
Comment (nicoburniske) [inline src/image.rs:28]: I'm inclined to use strongly typed `Resize` type here

File: src/image.rs [hunk 2]
```diff
@@ -9,17 +9,23 @@ use leptos_meta::Link;
...
     height: u32,
-    /// Resize image width, but will still maintain the same aspect ratio.
+    /// Resize image width.
     width: u32,
     /// Image quality. 0-100.
     #[prop(default = 75_u8)]
     quality: u8,
+    /// Filter type for the conversion : Nearest, Triangle, CatmullRom, Gaussian, Lanczos3
+    #[prop(default = "catmullrom")]
+    filter: &'a str,
```
Comment (nicoburniske) [inline src/image.rs:25]: Prefer strongly typed Enum here

File: src/image.rs [hunk 3]
```diff
@@ -9,17 +9,23 @@ use leptos_meta::Link;
 /// Image component for rendering optimized static images.
 /// Images MUST be static. Will not work with dynamic images.
 #[component]
-pub fn Image(
+pub fn Image<'a>(
```
Comment (nicoburniske) [inline src/image.rs:12]: Remove lifetime after changing `&str` to enums

**Review by nicoburniske — CHANGES_REQUESTED** [review]
Hey thanks for doing this. I am very busy these days and if you're interested in helping maintain this crate lmk. 

Minor comments. I prefer strongly typed enum values rather than  parsing raw strings and then using `.unwrap_or_default()`

I can get this merged after

File: src/optimizer.rs [hunk 4]
```diff
@@ -373,9 +454,9 @@ where
     P: AsRef<std::ffi::OsStr>,
 {
     match std::path::Path::new(&path).parent() {
-        Some(p) if (!(p).exists()) => std::fs::create_dir_all(p),
-        Some(_) => Result::Ok(()),
-        None => Result::Ok(()),
+        Some(p) if !p.exists() => std::fs::create_dir_all(p),
```
Comment (nicoburniske) [inline src/optimizer.rs:457]: also this should be tokio::fs

**Comment by SoapyDev** [comment]
Will redo the pull request, I see alot of commits that where not meant to be in it.

**Pull request closed without merging**